nimutils/awsclient.nim

```diff
@@ -149,20 +149,23 @@ proc request*(client: var AwsClient, params: Table, headers: HttpHeaders = newHt
   # Add signing key caching so we can skip a step
   # utilizing some operator overloading on the create_aws_authorization proc.
   # if passed a key and not headers, just return the authorization string; otherwise, create the key and add to the headers
+  var authHeaders = newHttpHeaders()
   if client.key_expires <= getTime():
     client.scope.date = getAmzDateString()
-    client.key = create_aws_authorization(client.credentials, req, headers.table, client.scope)
+    client.key = create_aws_authorization(client.credentials, req, authHeaders.table, client.scope)
     client.key_expires = getTime() + initTimeInterval(minutes = 5)
   else:
-    let auth = create_aws_authorization(client.credentials.id, client.key, req,
-        headers.table, client.scope)
+    let auth = create_aws_authorization(client.credentials.id, client.key, req, authHeaders.table, client.scope)
     headers.add("Authorization", auth)
```

Comment:
Should this be changed to the local `requestHeaders` as well? And what of the `headers` variable in the function prototype?

Reply:
:+1:  ahh cool. I meant the variable you're defining on line 152. 

It also looks like the `headers` variable from the function prototype is no longer used anywhere? 

from line 108: 

```
proc request*(client: var AwsClient, params: Table, headers: HttpHeaders = newHttpHeaders()):
```

Reply:
L152 defines `authHeaders` var, not sure where you are getting `requestHeaders` from

https://github.com/crashappsec/nimutils/blob/66995639828f0b79eaeb1c58ca2b24e84cc7be30/nimutils/awsclient.nim#L152

and no `headers` are still used a little below to copy provided headers into the `authHeaders` variable

https://github.com/crashappsec/nimutils/blob/66995639828f0b79eaeb1c58ca2b24e84cc7be30/nimutils/awsclient.nim#L161-L162

Reply:
:facepalm: sorry yeah I meant `authHeaders`.